taxcalc/calculate.py

```diff
@@ -58,6 +58,9 @@ def __init__(self, parameters, records, **kwargs):
         self._records = (records if not isinstance(records, str) else
                          Records.from_file(records, **kwargs))
 
+        while self._records.current_year < self._parameters.current_year:
```

Comment:
@TJ, could you give an example of the situation you envision? 

One option is:

        if sync_years:
            print("You loaded data for "+str(self._records.current_year))
            while self._records.current_year < self._parameters.current_year:
                self._records.increment_year()
            print("Your data have been extrapolated to "+str(self._records.current_year))
        assert self._parameters.current_year == self._records.current_year

But I'm not sure this deals with your concern. 

Also, if `sync_years==False`, we are always going to fail the assertion at the end anyways.